irma-core/state-machine.js

```diff
@@ -11,70 +11,105 @@ module.exports = class StateMachine {
     this._disabledTransitions = [];
   }
 
+  /* Deprecated
   currentState() {
     return this._state;
-  }
+  } *
 
+  /* Deprecated
   isEndState() {
     return this._inEndState;
-  }
+  } */
 
+  /* Deprecated
   isValidTransition(transition) {
     if (this._inEndState || this._disabledTransitions.includes(transition))
       return false;
     return transitions[this._state][transition] != undefined;
-  }
+  } */
```

Comment:
I'd like to fully remove these methods, since these ones are the main cause for race conditions. Devs should use the `selectTransition` method instead. This implies that developers that have custom plugins and use these functions have to update this right away. This forces them to fix potential mistakes.

Do you all agree with this approach or do you think it is better to keep them with a deprecation warning and remove them in a later update?

Reply:
I know one user that once asked questions on Slack that he is using transitions to force an abort when the session gets cancelled, but he doesn't use these three functions, so that would go fine. I'd also be in favour of removing these three ones with immediate effect.

Reply:
Since we didn't get any comments in Slack, we for now assume they can be removed immediately.